test/sequential/test-http-keep-alive-large-write.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const serverTimeout = common.platformTimeout(500);
+let offsetTimeout = common.platformTimeout(100);
+let socketHandle;
+let writeSize = 3000000;
+let writtenOffset = 0;
+let didReceiveData = false;
+
+const server = http.createServer(common.mustCall((req, res) => {
+  const content = Buffer.alloc(writeSize, 0x44);
+
+  res.writeHead(200, {
+    'Content-Type': 'application/octet-stream',
+    'Content-Length': content.length.toString(),
+    'Vary': 'Accept-Encoding'
+  });
+
+  socketHandle = res.socket._handle;
+  res.write(content);
+  res.end();
+}));
+server.setTimeout(serverTimeout);
+server.on('timeout', () => {
+  assert.strictEqual(didReceiveData, false, 'Should not timeout');
+});
+
+server.listen(0, common.mustCall(() => {
+  http.get({
+    path: '/',
+    port: server.address().port
+  }, common.mustCall((res) => {
+    const resume = () => res.resume();
+    let receivedBufferLength = 0;
+    let firstReceivedAt;
+    res.on('data', common.mustCallAtLeast((buf) => {
+      if (receivedBufferLength === 0) {
+        writtenOffset = writeSize - socketHandle.writeQueueSize;
+        didReceiveData = false;
+        firstReceivedAt = Date.now();
+      }
+      receivedBufferLength += buf.length;
+      if (receivedBufferLength >= writtenOffset) {
```

Comment:
If I understand this correctly this checks if we received more than 50%? If that is right could you do:
```js
const isHalfRecived = receivedBufferLength >= writtenOffset;
```
And comment why we need 50%?

Reply:
Not quite. Basically, by the time the first `data` event fires there has been a certain amount of stuff written. We get that amount and then consume it, then we call `pause`. This allows us to mimic a slow client. Then we wait longer than the timeout, _onTimeout fires and checks whether it had a chance to write anything or the timeout is real. Then we continue and do this as many times as necessary until the data is all consumed.

Reply:
Ment to delete this comment (figured out the `writeSizeOnFirstRead` thing).
But if you decide to keep the var name a comment would be great.